src/collective/siteimprove/browser/viewlets.py

```diff
@@ -13,9 +12,17 @@ def update(self):
         super(SiteimproveJavascriptViewlet, self).update()
 
         self.token = None
+        self.recheck = False
 
         # lookup token
         registry = getUtility(IRegistry)
         siteimprove_registry = registry.forInterface(ISiteimproveSchema, False)
         if siteimprove_registry and siteimprove_registry.token:
-            self.token = siteimprove_registry.token
\ No newline at end of file
+            self.token = siteimprove_registry.token
+
+        session_manager = self.context.session_data_manager
```

Comment:
I really don't like using zope sessions for this, they cause potential issues with load-balancing and make ConflictErrors more likely. Could we use a cookie instead, which we expire/delete after the recheck is sent?